tpp-app/src/services/SettingsService.js

```diff
@@ -115,43 +116,165 @@ import { TRACK_SYMPTOMS, REMINDERS, LOG_PERIOD_FREQ, LOG_SYMPTOMS_FREQ } from '.
         }
     });
 
-
-
-    export const POSTRemindLogPeriodAdvanceDays = async (advanceDays) => new Promise(async (resolve, reject) => {
+    /**
+     * Posts the frequency to send log period reminders
+     * @param {string} advanceDays how many days in advance of period to send reminder
+     * @returns a promise resolving when the post operation is complete
+     */
+    export const POSTRemindLogPeriodFreq = async (advanceDays) => new Promise(async (resolve, reject) => {
         try {
-            await AsyncStorage.setItem(REMINDERS)
+            await AsyncStorage.setItem(REMINDERS.LOG_PERIOD_DAYS, JSON.stringify(advanceDays)).then(() => {
+                console.log("Posted the number of days in advance to send log period reminder");
+                resolve();
+            }).catch((e) => {
+                console.log(JSON.stringify(e));
+                reject("Unable to post number of days in advance to send log period reminder ");
+            })
             
-        } catch(e) {
-
+        } catch (e) {
+    console.log(`POSTRemindLogPeriodFreq error: ${JSON.stringify(e)}`)
+            reject();        
         }
     });
 
-    export const GETRemindLogPeriodAdvanceDays = async () => new Promise(async (resolve, reject) => {
-
+    /**
+     * Retrieves the frequency to send log period reminders 
+     * @returns a promise resolving in a string representing the number of days
+     */
+    export const GETRemindLogPeriodFreq= async () => new Promise(async (resolve, reject) => {
+        try {
+            value = await AsyncStorage.getItem(REMINDERS.LOG_PERIOD_DAYS).then((value) => {
+                        console.log("Retrieved the number of days in advance to send log period reminder");
+                        resolve(value);
+                    }).catch((e) => {
+                        console.log(JSON.stringify(e));
+                        reject("Unable to retrieve number of days in advance to send log period reminder");
+                    })
+        } catch (e) {
+            console.log(`POSTRemindLogPeriodFreq error: ${JSON.stringify(e)}`)
+            reject();
+        }
     });
 
-    export const POSTRemindLogPeriodTime = async () => new Promise(async (resolve, reject) => {
 
+    /**
+     * Posts the time (exp: 1 am, 2 am) to send reminder to log period 
+     * @param {string} time string representing the time
```

Comment:
up to preference but i would have this param as an int. so 0 - 23 representing a 24 hours.  I think it's a lot easier to guard against in case of bad input string, but if you have a way to clean inputs in frontend so that it's guaranteed to receive an expected input here that would be fine too

Reply:
IDK use case for this func but also that makes it compatible directly with Date objs in JS which is added benefit